packages/gatsby-plugin-netlify/src/__tests__/build-headers-program.js

```diff
@@ -173,6 +173,27 @@ describe(`build-headers-program`, () => {
     ).toMatchSnapshot()
   })
 
+  it(`without manifest['pages-manifest']`, async () => {
+    const pluginData = await createPluginData()
+
+    // gatsby 2.9+ no longer has a pages-manifest key
+    delete pluginData.manifest[`pages-manifest`]
+
+    const pluginOptions = {
+      ...DEFAULT_OPTIONS,
+      mergeCachingHeaders: true,
+    }
+
+    await buildHeadersProgram(pluginData, pluginOptions, reporter)
+
+    expect(reporter.warn).not.toHaveBeenCalled()
+    const output = await fs.readFile(
+      pluginData.publicFolder(`_headers`),
+      `utf8`
+    )
+    expect(output).toMatchSnapshot()
```

Comment:
Can we add more specific assertion that checks directly if `/undefined` is not in the output?

Reply:
Assertion added.